### ISSUE TITLE
Unable to publish video track after connecting to room

### ISSUE THREAD

**Issue by akhil-kp-dasan** [twilio/twilio-video.js#954]
- [x] I have verified that the issue occurs with the latest twilio-video.js release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-video.js/blob/master/CHANGELOG.md).
 - [x] I reviewed the [Common Issues](https://github.com/twilio/twilio-video.js/blob/master/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
 - [x] I verified that the [Quickstart](https://github.com/twilio/video-quickstart-js) application works in my environment.
 - [x] I am not sharing any [Personally Identifiable Information (PII)](https://www.twilio.com/docs/glossary/what-is-personally-identifiable-information-pii)
       or sensitive account information (API keys, credentials, etc.) when reporting this issue.

**Code to reproduce the issue:**

```js
let config = {};

let localTracks = Twilio.Video.createLocalTracks();

config.networkQuality = true;
config.iceServersTimeout = 30000;
config.abortOnIceServersTimeout = true;
config.automaticSubscription = true;
config.tracks = localTracks;

let room = await Twilio.Video.connect(token, config);

room.localParticipant.on('trackPublished', () => console.log('track published')); //The track never gets published

let mediaStream = await navigator.mediaDevices.getDisplayMedia({video: true})
let screenTrack = mediaStream.getVideoTracks()
let screenShareTrack = Twilio.Video.LocalVideoTrack(screenTrack[0], {name: 'screenShareTrack'});
let publication = await room.localParticipant.publishTrack(screenShareTrack);
console.log(publication); //publication is not defined
```

**Expected behavior:**
Participant should be able to publish tracks after joining the room.

**Actual behavior:**
When publishing tracks after joining room the track doesn't get published. The `await room.localParticipant.publishTrack(track)` method doesn't publish the track nor does it throw any error.
This issue happens when any participant tries to screen share or switch cameras after joining the room.

Logs:
[twiliologs.txt](https://github.com/twilio/twilio-video.js/files/4481646/twiliologs.txt)

TODO

**Software versions:**

 - [x] Browser(s): Chrome 81
 - [x] Operating System: Windows 10
 - [x] twilio-video.js: 2.3.0
 - [x] Third-party libraries (e.g., Angular, React, etc.): Angular 6

**Comment by makarandp0** [comment]
Thank you for reporting this issue @akhil-kp-dasan 

I looked at the logs you provided for the room (RM01b3ee0e4deb9571df4d16598dae1c34) and I do see some weirdness in the message sent after screenshare track was added by (PAf3e9c10f0cbf4f1fb312d7c77263ae8c)

I noticed that the logs attached were not complete. Can you see if you can obtain full logs? Any other information you can provide to help repro this issue would be very helpful as well. I am going to investigate this further, we are tracking a similar issue internally as (JSDK-2760)

Thanks,
Makarand

**Comment by akhil-kp-dasan** [comment]
I have attached the logs generated after performing following steps:
1) Connect both participants to room
2) Start screen share from participant 1
3) Wait 2 minutes
4) Stop screen share

[screen_share_participant1.log](https://github.com/twilio/twilio-video.js/files/4491445/screen_share_participant1.log)
[screen_share_participant2.log](https://github.com/twilio/twilio-video.js/files/4491446/screen_share_participant2.log)

Let me know if there is any other information I can provide to help identify the issue.

**Comment by manjeshbhargav** [comment]
Hi @akhil-kp-dasan ,

From my preliminary analysis, it looks like you are hitting a bug that we are working to fix this sprint. It is most likely due to a timing issue between how soon you publish the screen share track after connecting to the Room. If you're use case demands that screen share be available as soon as you join the Room, I recommend passing the screen share track in `ConnectOptions.tracks` along with your audio and video tracks as a workaround until we fix the bug.
```js
const localTracks = await Twilio.Video.createLocalTracks();
const screenStream = await navigator.mediaDevices.getDisplayMedia({ video: true });
const screenTrack = new Twilio.Video.LocalVideoTrack(screenStream.getVideoTracks()[0]);

localTracks.push(screenTrack);

const room = await Twilio.Video.connect('token', {
  name: 'my-room',
  tracks: localTracks
});
```
Please let me know if this works for you.

Thanks,

Manjesh Malavalli
JSDK Team

**Comment by akhil-kp-dasan** [comment]
It looks like the screen share track does get published if I add it to the connection config before connecting but unfortunately we can't use this workaround on production as we only want to start screen sharing when the participant decides to do so and not automatically upon connecting to the room.

**Comment by manjeshbhargav** [comment]
Hi @akhil-kp-dasan ,

What you are looking for should work. But in the code that you shared with me, you are publishing the screen track as soon as you join the Room. While we work to fix this use case, you can still get the screen share track to work if you publish it only after the user clicks on a button.

Please try it out and let me know if this works.

Thanks,

Manjesh Malavalli
JSDK Team

**Comment by akhil-kp-dasan** [comment]
We already have a button that the user has to click to start screen sharing. In the code sample I shared I removed those parts to simplify it.

Regarding your comment about this being a timing issue, so if I wait some time after joining the room and then start screen sharing would the track get published? I have tried waiting 10 minutes after joining the room and before starting screen share and the screen share track is still not getting published.
